lib/report.js

```diff
@@ -158,7 +160,158 @@ class ConciseReporter extends Reporter {
   }
 }
 
+class TapReporter extends Reporter {
+  constructor(options) {
+    super(options);
+    this.successful = 0;
+    this.failed = 0;
+    this.skipped = 0;
+    this.failedTests = [];
+    this.testNumeration = [1];
```

Comment:
Could you elaborate on what you use `this.testNumeration` for. 

If it's to count and distinguish tests then each `Test` have `test.id` which is unique across all Parent tests and each subtest of a parent also has an Id which starts from 0.

So if we have these tests `Parent1(subtest1, subtest2), Parent2, Parent3(subtest3)` then
* Parent1.id = 0
    * Parent1.subtest1.id = 0
    * Parent1.subtest2.id = 1
* Parent2.id = 1
* Parent3.id = 2
    * Parent3.subtest.3.id = 0

Peraps it will work if we use them?

As for the assertions (`results` array) then each of those already has id - index in the `results` array.

Reply:
The main problem in this case is that id index is given for asserts also, but as I understood TapReporter uses for tests only.

Reply:
Each assertion in a test has a unique index - an index in `test.results` array.